pkg/kubelet/volumemanager/volume_manager.go

```diff
@@ -208,12 +216,24 @@ type volumeManager struct {
 	desiredStateOfWorldPopulator populator.DesiredStateOfWorldPopulator
 }
 
-func (vm *volumeManager) Run(stopCh <-chan struct{}) {
+func (vm *volumeManager) Run(readyCh <-chan bool, stopCh <-chan struct{}) {
 	defer runtime.HandleCrash()
-	glog.Infof("Starting Kubelet Volume Manager")
 
-	go vm.reconciler.Run(stopCh)
 	go vm.desiredStateOfWorldPopulator.Run(stopCh)
+	glog.V(2).Infof("The desired_state_of_world populator starts")
+
+	// Add all sources ready check so that reconciler reconstruct process will start after
+	// desired state of world is populated with pod volume information from apiserver. Otherwise,
+	// reconciler reconstruct process may add incomplete volume information and cause confusion.
+	select {
+	case <-readyCh:
+		glog.V(4).Infof("Volume manager start after all source ready")
```

Comment:
Instead of blocking the entire volume manager, move this check to reconciler and around just the reconstruct loop so that normal pod mounting logic is not blocked.

As discussed in person, if all sources never become ready, then we simply should skip reconstruct loop indefinitely (maybe printing a warning after certain amount of time has passed).